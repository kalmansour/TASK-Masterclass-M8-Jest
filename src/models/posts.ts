export interface Post {
  id: number;
  name: string;
  email: string;
  address: string;
}

export type PostQuery =
  | {
      data: Post[];
      loading: false;
      error?: any;
    }
  | {
      data: undefined;
      loading: true;
      error: undefined;
    };
