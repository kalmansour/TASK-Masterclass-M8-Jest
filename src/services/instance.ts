import axios from "axios";

export const getApiUrl = (storage:Pick<Storage, "getItem"> = localStorage ): string => {
  
  if (process.env.NODE_ENV === 'development') {
    const url = storage.getItem('api') ?? process.env.REACT_APP_API_URL
    if (url){
      return url;
    }
  }
  return "https://api.example.com";
};

export const API = axios.create({
  baseURL: getApiUrl(),
});
