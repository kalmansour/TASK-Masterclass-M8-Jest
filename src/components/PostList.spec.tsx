import renderer from "react-test-renderer";

import { PostQuery } from "../models";
import { useGetPosts } from "../services";
import PostList from "./PostList";

jest.mock("../services/posts");

test("renders posts", () => {
    const query: PostQuery = {
      data: [
        {
        id: 1,
        name: "Satheesh",
        email: "Sat@mail.com",
        address: "South street"
      },
      {
        id: 2,
        name: "John",
        email: "john@mail.com",
        address: "North street"
      },
    ],
      loading:false,
    };
  
    const hook = useGetPosts as jest.Mock;
    hook.mockReturnValue(query);

    const tree = renderer.create(<PostList/>).toJSON();
    expect(tree).toMatchSnapshot();

});

