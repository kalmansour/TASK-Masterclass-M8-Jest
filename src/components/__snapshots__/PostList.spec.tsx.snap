// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders posts 1`] = `
<div>
  <h1>
    Posts
  </h1>
  <div>
    <h2>
      Satheesh
    </h2>
    <ul>
      <li>
        Email: 
        Sat@mail.com
      </li>
      <li>
        Address: 
        South street
      </li>
    </ul>
    <button>
      Go to post
    </button>
  </div>
  <div>
    <h2>
      John
    </h2>
    <ul>
      <li>
        Email: 
        john@mail.com
      </li>
      <li>
        Address: 
        North street
      </li>
    </ul>
    <button>
      Go to post
    </button>
  </div>
</div>
`;
